### PR TITLE
CATO-3540: Validation for E1030 and E1031.

### PR DESCRIPTION
**Pull request by tmikus**

### CODE REVIEW

File: src/main/scala/uk/gov/hmrc/ct/ct600e/v2/E1030.scala [hunk 1]
```diff
@@ -20,5 +20,6 @@ import uk.gov.hmrc.ct.box._
 import uk.gov.hmrc.ct.ct600e.v2.retriever.CT600EBoxRetriever
 
 case class E1030(value: Option[String]) extends CtBoxIdentifier("Claimer's name") with CtOptionalString with Input with ValidatableBox[CT600EBoxRetriever]{
-  override def validate(boxRetriever: CT600EBoxRetriever): Set[CtValidation] = validateStringAsMandatory("E1030", this)
+  override def validate(boxRetriever: CT600EBoxRetriever): Set[CtValidation] =
+    validateStringAsMandatory("E1030", this) ++ validateOptionalStringByLength("E1030", this, 2, 56) ++ validateOptionalStringByRegex("E1030", this, validNonForeignLessRestrictiveCharacters)
```
Comment (scottcutts) [inline src/main/scala/uk/gov/hmrc/ct/ct600e/v2/E1030.scala]: Is there a better name then `validNonForeignLessRestrictiveCharacters

File: src/main/scala/uk/gov/hmrc/ct/ct600e/v2/E1031.scala [hunk 2]
```diff
@@ -20,5 +20,6 @@ import uk.gov.hmrc.ct.box._
 import uk.gov.hmrc.ct.ct600e.v2.retriever.CT600EBoxRetriever
 
 case class E1031(value: Option[String]) extends CtBoxIdentifier("Claimer's status") with CtOptionalString with Input with ValidatableBox[CT600EBoxRetriever]{
-  override def validate(boxRetriever: CT600EBoxRetriever): Set[CtValidation] = validateStringAsMandatory("E1031", this)
+  override def validate(boxRetriever: CT600EBoxRetriever): Set[CtValidation] =
+    validateStringAsMandatory("E1031", this) ++ validateOptionalStringByLength("E1031", this, 2, 56) ++ validateOptionalStringByRegex("E1031", this, validNonForeignLessRestrictiveCharacters)
```
Comment (scottcutts) [inline src/main/scala/uk/gov/hmrc/ct/ct600e/v2/E1031.scala]: Here too

File: src/main/scala/uk/gov/hmrc/ct/ct600e/v2/E1030.scala (hunk 1 above)
Comment (tmikus) [inline src/main/scala/uk/gov/hmrc/ct/ct600e/v2/E1030.scala]: Well - I haven't created it and that's the name it's being referred to in all of the documents about XMLs and validation.

Comment (scottcutts) [inline src/main/scala/uk/gov/hmrc/ct/ct600e/v2/E1030.scala]: I see, it's the name in the RIM Rules, fair enough then. Bad name though :)

**Pull request merged**